### PR TITLE
Add PolyScheduler which mimics the caffe polynomial lr scheduler

### PR DESCRIPTION
**Pull request by bzcheeseman**
## Description ##
(Brief description on what this PR is about)
In reproducing a network originally defined in Caffe, I realized that MXNet had no equivalent LR scheduler. As a user of MXNet, I would like to be able to use such an LR scheduler, so I added one.

### Essentials ###
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- No tests exist currently for LR Scheduler(s), willing to add some if that is deemed useful.
- [ ] Code is well-documented: 
- Matches previous LR scheduler implementation, also willing to add some if deemed useful.
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Add `PolyScheduler` in `cpp-package/include/mxnet-cpp/lr_scheduler.h`

## Comments ##
- Matches quite closely the FactorScheduler implementation just above, but am happy to make other changes.

### CODE REVIEW

**Comment by piiswrong** [comment]
@bzcheeseman ping could fix the tests?

**Comment by lupesko** [comment]
@bzcheeseman - bouncing this to you as well, since lots of instability on the CI was fixed, your tests may be passing now. Please try again.
@marcoabreu - can you help with figuring out the docker test issues?

**Comment by bzcheeseman** [comment]
retest this please

**Comment by marcoabreu** [comment]
Sorry, I'm currently swamped with high priority tasks and I don't have time to review your pull request.

**Comment by lupesko** [comment]
@bzcheeseman you can always add a dummy commit. I'm not sure if there is a more elegant way...

**Comment by lebeg** [comment]
I'm really surprised by the errors in the builds. @bzcheeseman could you rebase your change on top of latest master?

**Comment by bzcheeseman** [comment]
I'm unsure why this PR is showing so many files changed after rebasing onto apache/master...from my side it just looks like 1 file. Should I close this and open a new PR?

**Comment by anirudhacharya** [comment]
@bzcheeseman yes either please clean up this PR by rebasing properly or open a new PR.

**Comment by bzcheeseman** [comment]
Closing in favor of new PR
